iam/src/providers/githubStarredRepoProvider.ts

```diff
@@ -0,0 +1,143 @@
+
+  // Exchange the code for an access token
+  const tokenRequest = await axios.post(
+    `https://github.com/login/oauth/access_token?client_id=${clientId}&client_secret=${clientSecret}&code=${code}`,
+    {},
+    {
+      headers: { Accept: "application/json" },
+    }
+  );
+
+  if (tokenRequest.status != 200) {
+    throw `Post for request returned status code ${tokenRequest.status} instead of the expected 200`;
+  }
+
+  const tokenResponse = tokenRequest.data as GithubTokenResponse;
+  return tokenResponse.access_token;
+};
+
+const verifyGithub = async (ghAccessToken: string): Promise<GithubFindMyUserResponse> => {
+  // Now that we have an access token fetch the user details
+  const userRequest = await axios.get("https://api.github.com/user", {
+    headers: { Authorization: `token ${ghAccessToken}` },
+  });
+
+  if (userRequest.status != 200) {
+    throw `Get user request returned status code ${userRequest.status} instead of the expected 200`;
+  }
+
+  return userRequest.data as GithubFindMyUserResponse;
+};
+
+const verifyUserGithubRepo = async (
+  userData: GithubFindMyUserResponse,
+  ghAccessToken: string
+): Promise<GithubUserRepoResponseData | boolean> => {
+  // Once access token is received, fetch user repo data
+  const repoRequest: GithubRepoRequestResponse = await axios.get(
+    `https://api.github.com/users/${userData.login}/repos?per_page=100`,
+    {
+      headers: { Authorization: `token ${ghAccessToken}` },
+    }
+  );
+
+  if (repoRequest.status != 200) {
+    throw `Get repo request returned status code ${repoRequest.status} instead of the expected 200`;
+  }
+
+  // Returns an object containing first instance of a user's repo if it has been starred
+  // by a user other than the repo owner, or the last checked repo with no stars
+  const checkUserRepoStars = async (): Promise<GithubUserRepoResponseData | boolean> => {
+    for (let i = 0; i < repoRequest.data.length; i++) {
+      const repo = repoRequest.data[i];
+      // Check if the GitHub user is the same as the repo owner
+      // and if the stargazer count is gt 1
+      if (userData.id === repo.owner.id && repo.stargazers_count > 1) {
+        return repo;
+        // Check if the GitHub user is the same as the repo owner and
+        // if their stargazer count equals 1
+      } else if (userData.id === repo.owner.id && repo.stargazers_count === 1) {
+        let stargazerData: StargazerData;
+        // GET the solo stargazer's data
+        const stargazerCheck = async (): Promise<[]> => {
+          const stargazerResponse = await axios.get(repo.stargazers_url);
+          const stargazerResponseData: [] = stargazerResponse.data;
+          return stargazerResponseData;
+        };
+
+        const stargazer = await stargazerCheck();
+        for (let i = 0; i < stargazer.length; i++) {
+          stargazerData = stargazer[i];
+        }
+
+        return stargazerData.id === repo.owner.id ? false : repo;
```

Comment:
Also I would simplify this:

```suggestion
        // GET the solo stargazer's data
        const stargazer = (await axios.get(repo.stargazers_url)).data;

        // Length should actually be 1 ... but we iterate just in case another star was added in the mean time
        for (let i = 0; i < stargazer.length; i++) {
          if (stargazer[i].id != repo.owner.id) {
            return true;
          }
        }
        return false;        
```